sources/Algo/src/Algo.cpp

```diff
@@ -164,35 +164,32 @@ LoadDefinitionAlgorithm::operator()(const NodePtr& node) {
 }
 /////////////////////////////////////////////////////////////////
 
-ControllerInterfaceDefinitionAlgorithm::ControllerInterfaceDefinitionAlgorithm(std::vector<HvdcLineDefinition>& hvdcLines) : hvdcLines_(hvdcLines) {}
+ControllerInterfaceDefinitionAlgorithm::ControllerInterfaceDefinitionAlgorithm(HvdcLineSet& hvdcLines) : hvdcLines_(hvdcLines) {}
 
 void
 ControllerInterfaceDefinitionAlgorithm::operator()(const NodePtr& node) {
   for (const auto& converter : node->converterInterfaces) {
     const auto& hvdcLine = converter.hvdcLine;
-    bool inserted;
-    std::tie(std::ignore, inserted) = hvdcLinesIds_.emplace(hvdcLine->id);
-    HvdcLineDefinition::Position position;
+    HvdcLineDefinition createdHvdcLine = HvdcLineDefinition(hvdcLine->id, hvdcLine->converterType, hvdcLine->converter1_id, hvdcLine->converter1_busId,
+                                                            hvdcLine->converter1_voltageRegulationOn, hvdcLine->converter2_id, hvdcLine->converter2_busId,
+                                                            hvdcLine->converter2_voltageRegulationOn, HvdcLineDefinition::Position::BOTH_IN_MAIN_COMPONENT);
+    auto it = hvdcLines_.find(createdHvdcLine);
+    bool alreadyInserted = it != hvdcLines_.end();
```

Comment:
you can check the iterator without using an local variable

Reply:
It is for the clarity of the code